spec/weak-types.html

```diff
@@ -0,0 +1,348 @@
+    </emu-clause>
+
+    <emu-clause id="sec-liveness">
+      <h1>Liveness</h1>
+
+      <p>For some set of objects _S_, a <dfn>hypothetical WeakRef-oblivious</dfn> execution with respect to _S_ is an execution whereby the abstract operation WeakRefDeref of a WeakRef whose referent is <ins>an unforgeable value that <emu-xref href="#sec-unforgeable-values">contains</emu-xref></ins> an element of _S_ always returns *undefined*.</p>
+
+      <emu-note>
+        WeakRef-obliviousness, together with liveness, capture two notions. One, that a WeakRef itself does not keep an <del>object</del><ins>unforgeable value</ins> alive. Two, that cycles in liveness does not imply that an <del>object</del><ins>unforgeable value</ins> is live. To be concrete, if determining _obj_'s liveness depends on determining the liveness of another WeakRef referent, <del>_obj2_</del><ins>_val2_</ins>, <del>_obj2's liveness</del><ins>the liveness of the objects _obj2_ such that _val2_ <emu-xref href="#sec-unforgeable-values">contains</emu-xref> _obj2_</ins> cannot assume _obj_'s liveness, which would be circular reasoning.
+      </emu-note>
+      <emu-note>
+        WeakRef-obliviousness is defined on sets of objects instead of individual objects to account for cycles. If it were defined on individual objects, then an object in a cycle will be considered live even though its Object value is only observed via WeakRefs of other <del>objects</del><ins>unforgeable values</ins> in the cycle.
+      </emu-note>
+      <emu-note>
+        Colloquially, we say that an individual object is live if every set of objects containing it is live.
+      </emu-note>
+
+      <p>At any point during evaluation, a set of objects _S_ is considered <dfn>live</dfn> if either of the following conditions is met:</p>
+
+      <ul>
+        <li>
+          Any element in _S_ is included in any agent's [[KeptAlive]] List.
+        </li>
+        <li>
+          There exists a valid future hypothetical WeakRef-oblivious execution with respect to _S_ that observes the Object value of any object in _S_.
+        </li>
+      </ul>
+      <emu-note>
+        The second condition above intends to capture the intuition that an object is live if its identity is observable via non-WeakRef means. An object's identity may be observed by observing a strict equality comparison between <del>objects</del><ins>unforgeable values</ins> or observing <del>the object</del><ins>an unforgeable value that <emu-xref href="#sec-unforgeable-values">contains</emu-xref> the object</ins> being used as key in a Map.</del>
+      </emu-note>
+      <emu-note>
+        <p>Presence of an <del>object</del><ins>unforgeable value</ins> in a field, an internal slot, or a property does not imply that the <del>object</del><ins>unforgeable value</ins> is live. For example if the <del>object</del><ins>unforgeable value</ins> in question is <ins>an object that is</ins> never passed back to the program, then it cannot be observed.</p>
```

Comment:
```suggestion
        <p>Presence of an <del>object</del><ins>unforgeable value</ins> in a field, an internal slot, or a property does not imply that the <del>object</del><ins>unforgeable value</ins> is live. For example if the <del>object</del><ins>unforgeable value</ins> in question is never passed back to the program, then it cannot be observed.<ins>Similarly, if an object _obj_ is not observed by the program, any unforgeable value that <emu-xref href="#sec-unforgeable-values">contains</emu-xref> the object _obj_ cannot be observed.</ins></p>
```

I think this addition is the crux of it to say that a record/tuple/box identity cannot be observed if the program no longer has the ability to observe any of the objects it contains (the obvious observation path being the record/tuple/box itself if that is still observable on its own).